contentcuration/contentcuration/assets/channelList/views/Channel/ChannelModal.vue

```diff
@@ -0,0 +1,705 @@
+                        <VExpansionPanelContent>
+                          <template v-slot:header>
+                            {{ $tr('more', {more: channelDetails.accessible_languages.length - 10}) }}
+                          </template>
+                          <VChip v-for="lang in channelDetails.accessible_languages.slice(10)" :key="lang">
+                            {{ lang }}
+                          </VChip>
+                        </VExpansionPanelContent>
+                      </VExpansionPanel>
+                    </VFlex>
+                  </VLayout>
+                  <VLayout v-if="channelDetails.includes.coach_content || channelDetails.includes.exercises" row wrap>
+                    <VFlex xs4>
+                      {{ $tr('instructor_resources') }}
+                    </VFlex>
+                    <VFlex v-if="channelDetails.includes.coach_content" xs4>
+                      <VChip :title="$tr('role_description')">
+                        {{ $tr('coach_content') }}
+                      </VChip>
+                    </VFlex>
+                    <VFlex v-if="channelDetails.includes.exercises" xs4>
+                      <VChip :title="$tr('role_description')">
+                        {{ $tr('assessments') }}
+                      </VChip>
+                    </VFlex>
+                  </VLayout>
+                  <VLayout v-if="channelDetails.sample_nodes" row wrap>
+                    <VFlex xs6>
+                      {{ $tr('preview') }}
+                      <VLayout v-if="channelDetails.simple_pathway" row wrap>
+                        <VFlex xs12>
+                          <a
+                            v-for="node in channelDetails.simple_pathway"
+                            :key="node.node_id"
+                            :href="nodeLink(node.node_id)"
+                            target="_blank"
+                          >
+                            <VChip>
+                              <VIcon>{{ node.kind | kindIcon }}</VIcon>
+                              {{ node.title }}
+                            </VChip>
+                          </a>
+                        </VFlex>
+                      </VLayout>
+                      <VLayout row wrap justifyCenter>
+                        <VFlex v-for="node in channelDetails.sample_nodes" :key="node.node_id" xs3>
+                          <a
+                            :href="nodeLink(node.node_id)"
+                            target="_blank"
+                          >
+                            <VImg contain :src="node.thumbnail" />
+                            <h5>{{ node.title }}</h5>
+                          </a>
+                        </VFlex>
+                      </VLayout>
+                      <VLayout v-if="channelDetails.tags" row wrap>
+                        <VFlex xs12>
+                          <!-- Tag cloud will be inserted here -->
```

Comment:
Found a library that might be useful here: https://github.com/SeregPie/VueWordCloud In particular, it looks like it's much better with responsiveness, which means you can do away with a lot of the logic in the wordcloud generator code